packages/toolpad-app/src/toolpadDataSources/function/client.tsx

```diff
@@ -161,66 +161,51 @@ function QueryEditor({
   const handleLogClear = React.useCallback(() => setPreviewLogs([]), []);
   const handleHarClear = React.useCallback(() => setPreviewHar(createHarLog()), []);
 
-  const handleCommit = React.useCallback(
-    () =>
-      onChange({
-        ...input,
-        query: {
-          ...input.query,
-          module: tryFormat(input.query.module),
-        },
-      }),
-    [onChange, input],
-  );
-
+  const handleCommit = React.useCallback(() => onCommit?.(), [onCommit]);
   useShortcut({ code: 'KeyS', metaKey: true }, handleCommit);
 
-  const isDirty = input !== value;
-
   return (
-    <QueryEditorShell onCommit={handleCommit} isDirty={isDirty}>
-      <SplitPane split="vertical" size="50%" allowResize>
-        <SplitPane split="horizontal" size={85} primary="second" allowResize>
-          <QueryInputPanel onRunPreview={handleRunPreview}>
-            <Box sx={{ flex: 1, minHeight: 0 }}>
-              <TypescriptEditor
-                value={input.query.module}
-                onChange={(newValue) =>
-                  setInput((existing) => ({ ...existing, query: { module: newValue } }))
-                }
-                extraLibs={extraLibs}
-              />
-            </Box>
-          </QueryInputPanel>
-
-          <Box sx={{ p: 2, height: '100%', overflow: 'auto' }}>
-            <Typography>Parameters</Typography>
-            <ParametersEditor
-              value={input.params}
-              onChange={(newParams) => setInput((existing) => ({ ...existing, params: newParams }))}
-              globalScope={globalScope}
-              liveValue={paramsEditorLiveValue}
+    <SplitPane split="vertical" size="50%" allowResize>
+      <SplitPane split="horizontal" size={85} primary="second" allowResize>
+        <QueryInputPanel onRunPreview={handleRunPreview}>
+          <Box sx={{ flex: 1, minHeight: 0 }}>
+            <TypescriptEditor
+              value={input.query.module}
+              onChange={(newValue) =>
```

Comment:
How do we usually decide whether we should wrap handler with `useCallback` (i.e. `handleCommit`) or can just pass them inline?

Reply:
You're encouraged to always use `useCallback` (or our own custom `useEvent`). Though I don't want to make this a hard rule.

Mind you that the only thing that changed to this code is indentation. I didn't do a review of all the code that I reindented. I recommend reviewing with _whitespace only_ enabled. About 1/3 to half of the code has been reindented.